keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/constraint/BakeConstraintEvaluator.kt

```diff
@@ -0,0 +1,73 @@
+package com.netflix.spinnaker.keel.bakery.constraint
+
+import com.netflix.frigga.ami.AppVersion
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.Locatable
+import com.netflix.spinnaker.keel.api.RegionSpec
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.artifacts.DebianArtifact
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.bakery.api.BakeConstraint
+import com.netflix.spinnaker.keel.clouddriver.ImageService
+import com.netflix.spinnaker.keel.constraints.ConstraintState
+import com.netflix.spinnaker.keel.constraints.StatefulConstraintEvaluator
+import com.netflix.spinnaker.keel.constraints.SupportedConstraintType
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import kotlinx.coroutines.runBlocking
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+
+class BakeConstraintEvaluator(
+  override val eventPublisher: ApplicationEventPublisher,
+  private val imageService: ImageService,
+  private val dynamicConfigService: DynamicConfigService,
+  repository: KeelRepository
+) : StatefulConstraintEvaluator<BakeConstraint>(repository) {
```

Comment:
`BakeConstraint` is a subtype of `Constraint` (but not `StatefulConstraint`) and the evaluator doesn't do anything with `ConstraintState`, so you may want to change this to subclass `ConstraintEvaluator` instead. 

If you leave this as a `StatefulConstraintEvaluator`, a `ConstraintState` db row is created in `PENDING` status, and you'll need to explicitly persist it to `PASS`. But since the constraint evaluator isn't triggering bakes and has no need to do things like track and monitor orca task id's, it's only checking if a condition is met, I think it is a better fit for the stateless interface.

Reply:
We should also make the combination of a stateless constraint type with the stateful evaluator type impossible.

Reply:
Do we actually need `StatefulConstraint` to exist?